CONTRIBUTING.md

```diff
@@ -0,0 +1,324 @@
+# Contributing Guide
```

Comment:
I think since the guide is quite long, it would be great to provide a small table of contents at the top. Something like what is described here. https://www.setcorrect.com/portfolio/work11/